jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -401,11 +442,12 @@ public JibContainerBuilder toContainerBuilder() throws IOException {
           "Failed to construct entrypoint because no files were added to the JavaContainerBuilder");
     }
 
-    JavaLayerConfigurations.Builder layerConfigurationsBuilder = JavaLayerConfigurations.builder();
+    Map<LayerType, LayerConfiguration.Builder> layerBuilders = new EnumMap<>(LayerType.class);
 
     // Add classes to layer configuration
     for (PathPredicatePair directory : addedClasses) {
-      layerConfigurationsBuilder.addDirectoryContents(
+      addDirectoryContentsToLayer(
+          layerBuilders,
```

Comment:
How about having `layerBuilders` as a `final` field and not pass it around?

Reply:
Was worried about what happens if people call `toContainerBuilder()` more than once, but I guess I can just clear the field at the beginning of the method.

Reply:
Ah, that's a good point. Then I think I like the way it is now.